sim/midas/src/main/cc/simif_token_hashers.cc

```diff
@@ -0,0 +1,222 @@
+#include "simif_token_hashers.h"
+#include "simif.h"
+
+#include <iostream>
+#include <sstream>
+#include <fstream>
+
+/**
+ * The constructor for simif_token_hashers_t.
+ *
+ * @param [in] p A pointer to the parent instance of simif_t
+ */
+simif_token_hashers_t::simif_token_hashers_t(simif_t *p) : parent(p) {
```

Comment:
If it does not make sense for the pointer to be null, please use a reference instead.